source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -128,7 +128,10 @@ void EnvoyQuicClientConnection::setUpConnectionSocket(Network::ConnectionSocket&
       connection_socket.close();
     }
   }
-  if (!connection_socket.isOpen()) {
+  if (!connection_socket.isOpen() && connectionSocket().get() == &connection_socket) {
+    // Only close the connection if the connection socket is the current one. If it is a probing
+    // socket that isn't the current socket, do not close the connection upon failure, as the
+    // current socket is still usable.
```

Comment:
Is this a bug fix that we could make stand-alone, or is that not possible w/o using connect'd sockets?

Reply:
it was only triggered after adding connected sockets